src/lib/storage/frame_of_reference/frame_of_reference_encoder.hpp

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <array>
+#include <limits>
+#include <memory>
+
+#include "storage/base_column_encoder.hpp"
+
+#include "storage/frame_of_reference_column.hpp"
+#include "storage/value_column.hpp"
+#include "storage/value_column/value_column_iterable.hpp"
+#include "storage/vector_compression/vector_compression.hpp"
+#include "types.hpp"
+#include "utils/enum_constant.hpp"
+
+namespace opossum {
+
+class FrameOfReferenceEncoder : public ColumnEncoder<FrameOfReferenceEncoder> {
+ public:
+  static constexpr auto _encoding_type = enum_c<EncodingType, EncodingType::FrameOfReference>;
+  static constexpr auto _uses_vector_compression = true;
+
+  template <typename T>
+  std::shared_ptr<BaseEncodedColumn> _on_encode(const std::shared_ptr<const ValueColumn<T>>& value_column) {
+    const auto alloc = value_column->values().get_allocator();
+
+    static constexpr auto block_size = FrameOfReferenceColumn<T>::block_size;
```

Comment:
Is there a reason why this has to be a compile-time constant? Couldn‘t we have FOR columns with different block sizes?

Reply:
We could. We’d just have to figure out how to determine the optimal block size for a given column. If the block size wasn’t a compile-time constant, decompression would probably me a bit slower though.

Reply:
The encoding algorithm would also be more complicated.

Reply:
Again, maybe just add that as a comment.